flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterTest.java

```diff
@@ -353,13 +359,17 @@ public void declineCheckpoint(DeclineCheckpoint declineCheckpoint) {
             RpcCheckpointResponder rpcCheckpointResponder =
                     new RpcCheckpointResponder(jobMasterGateway);
             rpcCheckpointResponder.declineCheckpoint(
-                    jobGraph.getJobID(), new ExecutionAttemptID(), 1, userException);
+                    jobGraph.getJobID(), new ExecutionAttemptID(), 1, checkpointException);
 
             Throwable throwable =
                     declineCheckpointMessageFuture.get(
                             testingTimeout.toMilliseconds(), TimeUnit.MILLISECONDS);
-            assertThat(throwable, instanceOf(SerializedThrowable.class));
-            assertThat(throwable.getMessage(), equalTo(userException.getMessage()));
+            assertThat(throwable, instanceOf(CheckpointException.class));
+            Optional<Throwable> throwableWithMessage =
+                    ExceptionUtils.findThrowableWithMessage(throwable, userException.getMessage());
+            assertTrue(throwableWithMessage.isPresent());
+            assertThat(
+                    throwableWithMessage.get().getMessage(), equalTo(userException.getMessage()));
```

Comment:
Quick question concerning this test @Myasuka: What is this test testing after your changes? To me it looks as if we are testing that `new CheckpointException(CheckpointFailureReason.CHECKPOINT_DECLINED, userException);` correctly transports the `userException`.

Reply:
This test ensure the throwable contains message from `userException`.

Reply:
But why do we start a `JobMaster` in order to test this kind of behaviour. What we are effectively testing here is that the passed `CheckpointException` still contains the cause as it was constructed with. The test seems to be prohibitively expensive for such a trivial behavioral test.

Reply:
The test already existed before I modify. The main idea is to test the deserialized exception within JM side acts as expected from TM side, which is to verify FLINK-10419. Though it is a bit expensive, but I think we still need to verify it on JM side. Maybe we could simplify the test to avoid setting up some unnecessary components, but the kernel idea is worthy.

Reply:
Hmm to me it looks as if this test actually tests that `DeclineCheckpoint` sends the user exception in a serialized format. This should be testable w/o starting a JM and two `ActorSystems`.